insights/specs/default.py

```diff
@@ -948,6 +948,46 @@ def is_mod_loaded_for_ss(broker):
     sssd_config = simple_file("/etc/sssd/sssd.conf")
     subscription_manager_id = simple_command("/usr/sbin/subscription-manager identity")  # use "/usr/sbin" here, BZ#1690529
     subscription_manager_installed_product_ids = simple_command("/usr/bin/find /etc/pki/product-default/ /etc/pki/product/ -name '*pem' -exec rct cat-cert --no-content '{}' \;")
+
+    @datasource(Ps, HostContext)
+    def proc_smaps(broker):
+        ps = broker[Ps]
+        data = []
+        for p_id in ps:
+            pid_id = p_id["PID"]
+            try:
+                with open("/proc/{0}/smaps".format(pid_id), "r") as fp:
+                    swap = {}
+                    for line in fp.readlines():
+                        if not len(line):
+                            continue
+                        else:
+                            k, v = line.split(":", 1)
+                            if "00000000" in k or k == "VmFlags" or line[0].isdigit():
+                                continue
+                            elif k not in swap:
+                                swap[k] = int(v.split()[0])
+                            elif k in swap:
+                                swap[k] = swap[k] + int(v.split()[0])
+                            else:
+                                continue
+                    if swap:
+                        data.append(dict(pid=int(pid_id), det=swap))
+            except Exception:
+                continue
+
+        smaps_data = []
+        for proc in ps:
```

Comment:
Is this `loop` essential? I think it could be done inner the above loop of `L956`